File: LEETCODE-PROBLEMS/09 - Palindrome Number - Easy/PalindromeNumber.py
# Given an integer x, return true if x is a 
# palindrome, and false otherwise.

# the general approach is to use slicing
# for reading the string backwards to compare with input

class PalindromeNumber(object):
    def isPalindrome(self, x):
        if str(x) == str(x)[::-1]: return True
        return False
    
    def isPalindrome_evenMoreClean(self, x):
        return str(x)==str(x)[::-1]

File: LEETCODE-PROBLEMS/09 - Palindrome Number - Easy/PalindromeNumber.ts
/* Given an integer x, return true if x is a 
 *  palindrome, and false otherwise.
 */
class PalindromeNumber {

    /**
     * Returns true if x is palindrome
     * and returns false if not
     * @param x input integer
     * @returns boolean representation of palindrome status of x
     */
    public isPalindrome(x: number): boolean {
        let rev: string = x.toString().split("").reverse().join(""); // supposedly the ts equivalent of python slicing
        return rev===x.toString();
    }
}